framework/event_system/SystemWideEventPublisher.h

```diff
@@ -0,0 +1,27 @@
+#ifndef CHITECH_SYSTEMWIDEEVENTPUBLISHER_H
+#define CHITECH_SYSTEMWIDEEVENTPUBLISHER_H
+
+#include "event_system/EventPublisher.h"
+
+namespace chi
+{
+
+class SystemWideEventPublisher : public chi::EventPublisher
+{
+public:
+  static SystemWideEventPublisher& GetInstance();
+
+  SystemWideEventPublisher(const SystemWideEventPublisher&) =
+    delete; // Deleted copy constructor
+  SystemWideEventPublisher operator=(const SystemWideEventPublisher&) =
+    delete; // Deleted assignment operator
```

Comment:
```suggestion
  SystemWideEventPublisher(const SystemWideEventPublisher&) = delete;
  
  SystemWideEventPublisher 
  operator=(const SystemWideEventPublisher&) = delete;
```

Semantic, feel free to ignore